frontend/public/components/file-upload/file-upload-context.ts

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import * as _ from 'lodash';
+import { AlertVariant } from '@patternfly/react-core';
+import { FileUploadExtension, isFileUploadExtension } from '@console/plugin-sdk';
+import { useResolvedExtensions } from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
+import { NamespaceContext } from '@console/app/src/components/detect-namespace/namespace';
+import { useToast } from '@console/shared/src/components/toast';
+import { isFileSupported } from './file-upload-utils';
+
+export type FileUploadContextType = {
+  fileObj: File;
+  canDropFile: boolean;
+  setFileObj: (file: File) => void;
+};
+
+export const FileUploadContext = React.createContext<FileUploadContextType>({
```

Comment:
Move code from `public` into `console-app`. 
We should not be adding new capabilities into `public`.